### PR TITLE
KAFKA-3202: System test that changes message version on the fly

### PR DESCRIPTION
**Pull request by enothereska**
@becketqin @apovzner please have a look. @becketqin the test fails when the producer and consumer are 0.9.x and the message format changes on the fly.

### CODE REVIEW

**Comment by enothereska** [comment]
@ewencp if looks ok, could you please merge? Thanks.

**Comment by ijuma** [comment]
@enothereska I think we should test both 0.9 and 0.10 clients when it comes to changing the message version on the fly to 0.10, but only test 0.10 clients when changing the message version on the fly to 0.9.

File: tests/kafkatest/tests/message_format_change.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+        # Producer and consumer
+        self.producer_throughput = 10000
+        self.num_producers = 1
+        self.num_consumers = 1
+
+    def change_format(self, topic):
+        self.logger.info("First format change to 0.9.0")
+        self.kafka.alter_message_format(topic, str(LATEST_0_9))
+        time.sleep(1)
```
Comment (ewencp) [inline tests/kafkatest/tests/message_format_change.py]: Do we need `sleep` here and is it even sufficient to ensure we get the right behavior? I figure something like changing the format, then making sure at least one new message is produced + consumed would be the necessary and sufficient condition for proceeding to the subsequent change.

File: tests/kafkatest/tests/message_format_change.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+from kafkatest.services.kafka.version import LATEST_0_9, LATEST_0_10, TRUNK, KafkaVersion
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.services.console_consumer import ConsoleConsumer
+from kafkatest.utils import is_int
+from kafkatest.tests.produce_consume_validate import ProduceConsumeValidateTest
+from kafkatest.services.kafka import config_property
+import time
+
+#
+# This tests performs the following checks:
```
Comment (ewencp) [inline tests/kafkatest/tests/message_format_change.py]: This comment should move to a docstring, and probably on the test method itself so it makes it into the report.

**Comment by enothereska** [comment]
@ijuma Agreed on testing both clients, done now. Thanks.

**Comment by gwenshap** [comment]
LGTM

**Pull request closed without merging**